openmdao/solvers/solver.py

```diff
@@ -784,6 +816,77 @@ def solve(self, vec_names, mode, rel_systems=None):
         """
         raise NotImplementedError("class %s does not implement solve()." % (type(self).__name__))
 
+    def _solve(self):
+        """
+        Run the iterative solver.
+        """
+        maxiter = self.options['maxiter']
+        atol = self.options['atol']
+        rtol = self.options['rtol']
+        iprint = self.options['iprint']
+
+        self._mpi_print_header()
+
+        self._iter_count = 0
+        norm0, norm = self._iter_initialize()
+
+        self._norm0 = norm0
+
+        self._mpi_print(self._iter_count, norm, norm / norm0)
+
+        while self._iter_count < maxiter and norm > atol and norm / norm0 > rtol:
+            with Recording(type(self).__name__, self._iter_count, self) as rec:
+                self._single_iteration()
+                self._iter_count += 1
+                self._run_apply()
+                norm = self._iter_get_norm()
+                # With solvers, we want to record the norm AFTER the call, but the call needs to
+                # be wrapped in the with for stack purposes, so we locally assign  norm & norm0
```

Comment:
typo: 'in the with'

Reply:
I didn't write this comment, but it is very confusing and roundabout. Rewrote to simplify.